runltp-ng

```diff
@@ -141,11 +144,18 @@ if ($run) {
 	my %results;
 	$results{'sysinfo'} = utils::collect_sysinfo($backend);
 	my ($stats, $test_results) =
-	    utils::run_ltp($backend, $ltpdir, $run, $timeout,
-			   $include, $exclude);
+		utils::run_ltp($backend, $ltpdir, $run, $timeout,
+			       $include, $exclude);
 	$results{'tests'} = {'stats' => $stats, 'results' => $test_results};
-	results::writelog(\%results, "$logname.json");
-	results::writelog_html(\%results, "$logname.html");
+	if ($json_filter) {
+		my $log_data = results::json_filter($json_filter, \%results);
+		results::writelog($log_data, "$logname.json");
+		results::writelog_html($log_data, "$logname.html");
+	}
+	else {
+		results::writelog(\%results, "$logname.json");
+		results::writelog_html(\%results, "$logname.html");
+	}
```

Comment:
This is still unnecessarily complicated. Also I do not think that we should pass the filtered data to the writelog_html() function.

Why not just:

```
        my $json_data = \%results;

        if ($json_filter) {
                $json_data = results::json_filter($json_filter, $json_data);
        }

        results::writelog($json_data, "$logname.json");
        results::writelog_html(\%results, "$logname.html");
```

Or even simpler, why don't we pass the $json_filter to the writelog() function and call the `json_filter()` from there?

Reply:
> Or even simpler, why don't we pass the $json_filter to the writelog() function and call the `json_filter()` from there?

IMO this would be the best option

Reply:
I was not completely sure if the `writelog_html` should have different input. i will do it as the code above.